### PR TITLE
Connect company feed

### PR DESCRIPTION
**Pull request by TheBestTvarynka**

### CODE REVIEW

File: backend/core/src/main/java/com/feed_grabber/core/news/NewsService.java [hunk 1]
```diff
@@ -49,24 +49,26 @@ public Long getCountByCompanyId(UUID companyId) {
...
+		var news = News.builder()
+                .title(newsCreateDto.getTitle())
+                .type(newsCreateDto.getType())
+                .body(newsCreateDto.getBody());
+
+        if (newsCreateDto.getImageId() != null) {
+			var image = imageRepository.findById(newsCreateDto.getImageId())
                 .orElseThrow(NotFoundException::new);
+			news.image(image);
+		}
```
Comment (TheBestTvarynka) [inline backend/core/src/main/java/com/feed_grabber/core/news/NewsService.java:62]: The image is an optional thing. So it ca does not exist in the news.

File: frontend/src/components/CompanyFeedItemCreation/index.tsx [hunk 2]
```diff
@@ -33,36 +37,51 @@ const CompanyFeedItemCreation: FC<ConnectedFeedCreationProps & { match }> = ({
...
-          continue;
-        }
-        newLinks.push(await uploadFile(file));
+      if (!file.type.startsWith('image')) {
+        return;
       }
+      const imageId = await uploadFile(file);
       setIsUploading(false);
-      setItem({ ...item, images: newLinks as string[] });
+      console.log(imageId);
```
Comment (MaksymSlobodianyk) [inline frontend/src/components/CompanyFeedItemCreation/index.tsx]: Delete console.log please

File: frontend/src/components/CompanyFeedItemCreation/index.tsx [hunk 3]
```diff
@@ -33,36 +37,51 @@ const CompanyFeedItemCreation: FC<ConnectedFeedCreationProps & { match }> = ({
...
+    setImage(feedItem?.imageId);
   }, [feedItem]);
 
   const uploadFile = file => {
-    // upload image and return the link
-    return new Promise(resolve => resolve('https://i.imgur.com/4gfjYyI.png'));
+    const formData = new FormData();
+    formData.append('image', file);
+    return apiClient.post('/api/image', formData).then(res => {
+      console.log(res.data);
```
Comment (MaksymSlobodianyk) [inline frontend/src/components/CompanyFeedItemCreation/index.tsx]: Delete console.log please

File: frontend/src/components/CompanyFeedItemCreation/index.tsx [hunk 4]
```diff
@@ -33,36 +37,51 @@ const CompanyFeedItemCreation: FC<ConnectedFeedCreationProps & { match }> = ({
...
+      saveFeedItem(item);
+    } else {
+      console.log(item);
+      const newItem = {
+        title: item.title,
+        body: item.body,
+        type: item.type,
+        imageId: item.imageId
+      };
+      console.log({ newItem });
```
Comment (MaksymSlobodianyk) [inline frontend/src/components/CompanyFeedItemCreation/index.tsx]: Delete console.log please

File: frontend/src/components/CompanyFeedItemCreation/index.tsx [hunk 5]
```diff
@@ -33,36 +37,51 @@ const CompanyFeedItemCreation: FC<ConnectedFeedCreationProps & { match }> = ({
 
   const handleSubmit = () => {
     // here we send new item to backend
-    console.log(item);
-    saveFeedItem(item);
-    // history.goBack();
+    if (item.id) {
+      saveFeedItem(item);
+    } else {
+      console.log(item);
```
Comment (MaksymSlobodianyk) [inline frontend/src/components/CompanyFeedItemCreation/index.tsx]: Delete console.log please

File: frontend/src/components/MainPage/index.tsx [hunk 6]
```diff
@@ -29,7 +28,7 @@ interface IItem {
 interface IMainPageProps {
     questionnaireList: IQuestionnaireResponse[];
     reportsList?: IReportShort[];
-    newsList?: INewsItem[];
+//     newsList?: INewsItem[];
```
Comment (MaksymSlobodianyk) [inline frontend/src/components/MainPage/index.tsx]: Delete redudant commented code please

File: frontend/src/models/companyFeed/ICompanyFeedItem.ts [hunk 7]
```diff
@@ -1,9 +1,17 @@
...
+  body: string;
+  type: string;
+  createdAt: string;
+  imageId?: string;
+  user: IUserShort;
 }
 
+// export interface ICompanyFeedListState {
+//   pagination?: IPaginationInfo<ICompanyFeedItem>;
+//   isLoading: boolean;
```
Comment (MaksymSlobodianyk) [inline frontend/src/models/companyFeed/ICompanyFeedItem.ts]: Delete redudant commented code please

File: frontend/src/containers/Routing/index.tsx [hunk 8]
```diff
@@ -28,6 +28,7 @@ import RespondentReport from "../ReportPage/RespondentReport";
 import Profile from "../../components/Profile";
 import RedirectFormEditor from "../../components/RedirectFormEditor";
 import ErrorPage from "../ErrorPage";
+import NewsList from "../../components/NewsList";
 
 // just for demo
```
Comment (MaksymSlobodianyk) [inline frontend/src/containers/Routing/index.tsx]: Do we have something here that we need just for demo, or it is just an old comment?

**Review by MaksymSlobodianyk — CHANGES_REQUESTED** [review]

File: frontend/src/components/NewsList/index.tsx [hunk 9]
```diff
@@ -1,68 +1,87 @@
...
+  setPagination
+}) => {
+  const [t] = useTranslation();
+  return (
+    <div className={styles.newsItemContainer}>
+      <AccessManager staticPermission={Permissions.createPostsAndPolls}>
+        <Link to="/company/new">
+          <Button>{t('Add news')}</Button>
+        </Link>
+      </AccessManager>
```
Comment (TheBestTvarynka) [inline frontend/src/components/NewsList/index.tsx:64]: Only hr and company owner can create a new items for company feed.

**Review by MaksymSlobodianyk — APPROVED** [review]

**Pull request merged**